### ISSUE TITLE
error: wallet does not contain enough cardinal UTXOs, please add additional funds to wallet.

### ISSUE THREAD

**Issue by dude513** [casey/ord#1736]
Command used:

```
./ord wallet inscribe FILE --fee-rate 13
```

Output:

```
error: wallet does not contain enough cardinal UTXOs, please add additional funds to wallet.
```

Current UTXOs checked with `./ord wallet outputs`:

```
[
  {
    "output": "tx1:0",
    "amount": 10000
  },
  {
    "output": "tx2:0",
    "amount": 200000
  }
]
```

Does it mean `ord` looks for the first output ignoring the second with enough funds to spend?

**Comment by raphjaph** [comment]
If both outputs contain inscriptions then yes. Is that the case?

You can check whether that's true if the locations field in `ord wallet inscriptions` matches those outputs.

**Comment by dude513** [comment]
> `inscriptions`

No, I don't have any. Second output is the newly received funds.

**Comment by raphjaph** [comment]
Hmm that's weird. What does the `ord wallet balance` show?

**Comment by dude513** [comment]
> Hmm that's weird. What does the `ord wallet balance` show?

The balance was my both outputs combined. :)
I have made another deposit and still something doesn't work, might be a bug?

```
[
  {
    "output": "tx1:0",
    "amount": 10000
  },
  {
    "output": "tx2:0",
    "amount": 300000
  },
  {
    "output": "tx3:0",
    "amount": 200000
  }
]

```

Balance shows `"cardinal": 500000`.

**Comment by dude513** [comment]
I will create a new wallet and try to sweep funds into a single output, for this though I need to reindex all blocks with a new wallet, if it does fix my problem we might have a bug. 🐛

**Comment by veryordinally** [comment]
I have observed the same issue in one of my wallets. Enough cardinal UTXOs available, still same message. Will investigate.

**Comment by AlexLloyd0** [comment]
I'm seeing the same thing

```
$ ord wallet outputs
[
  {
    "output": "fcc8fd34ff6e0d8381db4aef8444ff3a4ab03199b4494938e9c1fe97577f21d6:0",
    "amount": 50000
  }
]
$ ord wallet inscriptions
[]
$ ord wallet balance
{
  "cardinal": 50000
}
$ RUST_LOG=debug ord wallet inscribe my_file.png
[2023-02-15T09:41:28Z INFO  ord::options] Connecting to Bitcoin Core RPC server at 127.0.0.1:8332/wallet/ord using credentials from `/home/alex/.bitcoin/.cookie`
[2023-02-15T09:41:28Z DEBUG bitcoincore_rpc] JSON-RPC request: getblockchaininfo []
[2023-02-15T09:41:28Z DEBUG bitcoincore_rpc] JSON-RPC request: getnetworkinfo []
[2023-02-15T09:41:28Z DEBUG bitcoincore_rpc] JSON-RPC request: getnetworkinfo []
[2023-02-15T09:41:28Z DEBUG bitcoincore_rpc] JSON-RPC request: listwallets []
[2023-02-15T09:41:28Z DEBUG bitcoincore_rpc] JSON-RPC request: listdescriptors []
[2023-02-15T09:41:28Z INFO  ord::index] Connecting to Bitcoin Core RPC server at 127.0.0.1:8332/wallet/ord using credentials from `/home/alex/.bitcoin/.cookie`
[2023-02-15T09:41:28Z DEBUG bitcoincore_rpc] JSON-RPC request: getblockcount []
[2023-02-15T09:41:28Z DEBUG bitcoincore_rpc] JSON-RPC request: getblockhash [776637]
[2023-02-15T09:41:28Z DEBUG bitcoincore_rpc] JSON-RPC error for getblockhash: RpcError { code: -8, message: "Block height out of range", data: None }
[2023-02-15T09:41:28Z DEBUG bitcoincore_rpc] JSON-RPC request: listunspent []
[2023-02-15T09:41:28Z DEBUG bitcoincore_rpc] JSON-RPC request: listlockunspent []
[2023-02-15T09:41:28Z DEBUG bitcoincore_rpc] JSON-RPC request: getrawchangeaddress ["bech32m"]
[2023-02-15T09:41:28Z DEBUG bitcoincore_rpc] JSON-RPC request: getrawchangeaddress ["bech32m"]
[2023-02-15T09:41:28Z DEBUG bitcoincore_rpc] JSON-RPC request: getrawchangeaddress ["bech32m"]
error: wallet does not contain enough cardinal UTXOs, please add additional funds to wallet.
```

**Comment by AlexLloyd0** [comment]
From what I can tell from adding a few debug logs:

src/subcommand/wallet/transaction_builder.rs:
```rust
  fn build_transaction(self) -> Result<Transaction> {
    self
      .select_outgoing()?         // <-- Removes entry from self.utxos
      .align_outgoing()
      .pad_alignment_output()?    // <-- Checks for *another* entry in self.utxos
      .add_value()?
      .strip_value()
      .deduct_fee()
      .build()
  }
```

I'm not sure why `pad_alignment_output` thinks that the output isn't big enough...

**Comment by dude513** [comment]
> I'm seeing the same thing

What branch/repo is that? @AlexLloyd0

**Comment by AlexLloyd0** [comment]
> > I'm seeing the same thing
> 
> What branch/repo is that? @AlexLloyd0 EDIT: I am on a `speedup-improvements` branch, andrewtoth's [repo](https://github.com/andrewtoth/ord).

`main` on `casey/ord`

**Comment by pavillag** [comment]
Any solution to this? I get the same error. 
![image](https://user-images.githubusercontent.com/30845498/219235008-00543063-f46a-4a53-82bd-0d63e55c2d03.png)

**Comment by aaacat6** [comment]
Is there a solution to this problem? I also encountered the same problem.

**Comment by yinchonging** [comment]
![企业微信截图_20230304170802](https://user-images.githubusercontent.com/2151476/222887788-5250eaa2-c0c4-4bdd-901d-58ac84d4565d.png)

Any solution to this? I get the same error!!!!

**Comment by aaacat6** [comment]
你的图片太大了,10kb左右试试
    
         
        
            
                        ***@***.***
            
        
     
    
  
    ---- Replied Message ----
  
  
    
      
        
         From 
        
        
            ***@***.***>
            
        
      
      
        
         Date 
        
        
        3/4/2023 17:15
        
      
      
        
         To 
        
        
         
          
            ***@***.***>
            
          
        
      
      
        
         Cc 
        
        
          
            ***@***.***>
            ,
          
          
            ***@***.***>
            
          
        
      
      
        
         Subject 
        
        
              Re: [casey/ord] error: wallet does not contain enough cardinal UTXOs, please add additional funds to wallet. (Issue #1736)
        
      
    
  




Any solution to this? I get the same error!!!!

**Comment by dude513** [comment]
> > Any solution to this? I get the same error. ![image](https://user-images.githubusercontent.com/30845498/219235008-00543063-f46a-4a53-82bd-0d63e55c2d03.png)
> 
> --fee-rate 10

It won't help.

**Comment by dude513** [comment]
I created a new wallet with 3 unspent outputs, this is a bug for sure. Anyone have some ideas?

**Comment by PicklErik21** [comment]
Same issue. created plenty utxos and it acts like none are available. most of them 20k sats big.

**Comment by PicklErik21** [comment]
Workaround for me was to send a bigger tx from my ord wallet to my sparrow wallet. Then made a "send to many"  with each tx being around 100k sats instead of 20k or 10k as before. Suddenly it worked and I was able to inscribe without error.

**Comment by rlisac** [comment]
i have already deposited about 20 dollars in btc for a normal etch order and still get the error although for example the etch should only cost 3 dollars in usd. do you already have information in this case what it could be?
